deepspeed/runtime/zero/parameter_offload.py

```diff
@@ -46,8 +46,17 @@ def _apply_to_tensors_only(module, functional, backward_function, outputs):
                                                   outputs[key])
         return outputs
 
-    elif type(outputs) is torch.Tensor:
-        return functional.apply(module, backward_function, outputs)
+    elif isinstance(outputs, torch.Tensor):
+        # this also applies to torch.Tensor's subclasses like torch.nn.parameter.Parameter
+        param = None
+        if is_zero_param(outputs):
+            param = outputs
+        touched_outputs = functional.apply(module, backward_function, outputs)
+
+        #using the temp to pass though the id of the original zero param
+        if param is not None:
+            touched_outputs.ds_param_alias = param
+        return touched_outputs
```

Comment:
Do we need to worry that this will now introduce an overhead for all tensors while we are dealing with an edge case?

I'm not sure if this is any faster since the copy has been made already

```suggestion
        param = None
        if is_zero_param(outputs):
            param = outputs
        touched_outputs = functional.apply(module, backward_function, outputs)

        #using the temp to pass though the id of the original zero param
        if param is not None and not is_zero_param(touched_outputs):
            touched_outputs.ds_param_alias = param
        return touched_outputs
```